src/client/camera.cpp

```diff
@@ -86,6 +86,36 @@ Camera::~Camera()
 	m_wieldmgr->drop();
 }
 
+void Camera::notifyFovChange()
+{
+	LocalPlayer *player = m_client->getEnv().getLocalPlayer();
+	assert(player);
+
+	PlayerFovSpec spec = player->getFov();
+
+	// If server-sent FOV is a multiplier, multiply
+	// it with m_cache_fov instead of overriding
+	m_server_sent_fov = spec.is_multiplier ? m_cache_fov * spec.fov : spec.fov;
+
+	// If an FOV transition is already active, make
+	// current FOV as the start of the new transition
+	if (m_fov_transition_active)
+		m_old_fov_degrees = m_curr_fov_degrees;
+
+	if (spec.transition_time > 0.0f)
+		m_fov_transition_active = true;
+
+	// If FOV smooth transition is active, initialize required variables
+	if (m_fov_transition_active) {
+		m_transition_time = spec.transition_time;
+		if (m_old_fov_degrees == 0.0f)
+			m_old_fov_degrees = m_cache_fov;
```

Comment:
Isn't `m_cache_fov == m_curr_fov_degrees`, which would also make the check in L102-103 unnecessary?

Reply:
`m_cache_fov` is strictly only used to store the client's default FOV (defined by the `fov` setting). This won't change throughout the game session (until we add support for runtime FOV changes at least :P).

`m_server_sent_fov` (this could be renamed to `m_target_fov` or smth. which describes its purpose better) holds the final/target FOV value if the server enforces a certain FOV (directly or via a multiplier).

`m_curr_fov_degrees` on the other hand, contains the real-time FOV value of the client - this value transitions from `m_old_fov_degrees` to:

- `m_server_sent_fov`, if the server sends a non-zero FOV value.
- `m_cache_fov`, if the server sends a zero FOV value.

Reply:
Let's take L102-L103 as an example:

`m_old_fov_degrees` would be equal to the target FOV of the previous transition, and is the starting point of the current/on-going transition. Meanwhile, `m_curr_fov_degrees` corresponds to the FOV of the client in real-time and is continuously increased by delta time until it reaches the target FOV of the current transition ( `== m_server_sent_fov`).

Now since the client is notified of a new FOV transition, `m_old_fov_degrees` (the starting point of a transition) is set to `m_curr_fov_degrees` (the current FOV) and the new transition starts from where the current transition stops. Once the transition is over, `m_old_fov_degrees` would be set to `m_curr_fov_degrees` (which is equal to `m_server_sent_fov`, or the server-sent target FOV).

Hope this makes sense. :)